index.bs

```diff
@@ -1772,6 +1963,278 @@ contexts that have active documents; navigation can also cause contexts to
+   <dd>
+      <pre class="cddl local-cddl">
+        BrowsingContextDomContentLoadedEvent = {
+         method: "browsingContext.domContentLoaded",
+         params: BrowsingContextNavigationInfo
+       }
+      </pre>
+   </dd>
+</dl>
+The [=remote end event trigger=] is:
+
+
+<div algorithm="remote end event trigger for browsingContext.domContentLoaded">
+
+Define the following [=dom load complete steps=] with |context| and |navigation status|:
+
+ 1. If the [=current session=] is null, return.
+
+ 1. Let |context id| be the [=browsing context id=] for |context|.
+
+ 1. Let |navigation id| be |navigation status|'s id.
+
+ 1. Let |url| be |navigation status|'s url.
+
+ 1. Let |params| be a [=map=] matching the
+    <code>BrowsingContextNavigationInfo</code> production, with the
+    <code>context</code> field set to |context id|, the <code>navigation</code>
+    field set to |navigation id|, and the <code>url</code> field set to the
+    result of the [=URL serializer=] given |url|.
+
+ 1. Let |body| be a [=map=] matching the
+     <code>BrowsingContextDomContentLoadedEvent</code> production, with the
+     <code>params</code> field set to |params|.
+
+ 1. Let |related browsing contexts| be a set containing |context|.
+
+ 1. [=Emit an event=] with |body| and |related browsing contexts|.
+
+ 1. [=Resume=] with "<code>DOMContentLoaded</code>", |navigation id|, and
+    |navigation status|.  </div>
+
+#### The browsingContext.load Event #### {#event-browsingContext-load}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+        BrowsingContextLoadEvent = {
+         method: "browsingContext.load",
+         params: BrowsingContextNavigationInfo
+       }
+      </pre>
+   </dd>
+</dl>
+The [=remote end event trigger=] is:
+
+
+<div algorithm="remote end event trigger for browsingContext.load">
+
+Define the following [=load complete steps=] with |context| and |navigation status|:
+
+ 1. If the [=current session=] is null, return.
+
+ 1. Let |context id| be the [=browsing context id=] for |context|.
+
+ 1. Let |navigation id| be |navigation status|'s id.
+
+ 1. Let |url| be |navigation status|'s url.
+
+ 1. Let |params| be a [=map=] matching the
+    <code>BrowsingContextNavigationInfo</code> production, with the
+    <code>context</code> field set to |context id|, the <code>navigation</code>
+    field set to |navigation id|, and the <code>url</code> field set to the
+    result of the [=URL serializer=] given |url|.
+
+1. Let |body| be a [=map=] matching the <code>BrowsingContextLoadEvent</code>
+     production, with the <code>params</code> field set to |params|.
+
+ 1. Let |related browsing contexts| be a set containing |context|.
+
+ 1. [=Emit an event=] with |body| and |related browsing contexts|.
+
+ 1. [=Resume=] with "<code>load</code>", |navigation id| and
+    |navigation status|.
+
+</div>
+
+#### The browsingContext.downloadWillBegin Event #### {#event-browsingContext-downoadWillBegin}
```

Comment:
On the naming, will it begin, or has it started? We have `downloadWillBegin` but "download started".

I think it's actually "has begun and may already have finished by now", so "download started" seems right.

Reply:
I'll take it :)